sonar-slang-plugin/src/test/java/com/sonarsource/slang/plugin/SyntaxHighlighterTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.plugin;
+
+import com.sonarsource.slang.kotlin.InputFileContext;
+import com.sonarsource.slang.parser.SLangConverter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.highlighting.TypeOfText;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.COMMENT;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.CONSTANT;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.KEYWORD;
+import static org.sonar.api.batch.sensor.highlighting.TypeOfText.STRING;
+
+public class SyntaxHighlighterTest {
+
+  private SyntaxHighlighter highlightingVisitor = new SyntaxHighlighter();
+
+  private SensorContextTester sensorContext;
+
+  private SLangConverter parser = new SLangConverter();
+
+  private DefaultInputFile inputFile;
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Before
+  public void setUp() {
+    sensorContext = SensorContextTester.create(tempFolder.getRoot());
+  }
+
+  private void highlight(String sqlCode) throws IOException {
```

Comment:
`sqlCode` ?
I have a good hint on which analyzer you inspired from for this part :)